### PR TITLE
Allow node to enter loop() if transport is NOK

### PR DESCRIPTION
**Pull request by tekka007**
WIP:
- allow node to enter loop() if transport is not fully initialized
- waiting time until transport ready: MY_TRANSPORT_WAIT_READY_MS (default 0 = no timeout = node does not enter main loop() without an established uplink)
- move node transport config (ID, parentNodeID, distance to GW) to transport layer
- remove MY_TRANSPORT_DONT_CARE_MODE
- rename typedefs / reshuffle core functions

### CODE REVIEW

**Comment by tekka007** [comment]
Jenkins, build examples please

**Comment by tekka007** [comment]
Jenkins, build examples please

**Comment by tekka007** [comment]
Jenkins, build examples please

**Comment by pepeEL** [comment]
When we can use it on MySensors Gateway and Nodes ?

**Comment by pepeEL** [comment]
When we can use it on MySensors Gateway and Nodes ?

And what is MY_TRANSPORT_RELAXED ?

**Comment by tekka007** [comment]
@pepeEL Please feel free to test this PR and provide feedback. This feature will be included in the upcoming 2.0.1 release.

**Comment by pepeEL** [comment]
Sorry but i am begginer. Can you explaine me step by step how i can test it on MySensors Gateway Serial and on my nodes ? Please

**Comment by pepeEL** [comment]
When will be available version 2.1.0 ?

**Comment by tekka007** [comment]
@pepeEL 2.1.0 will be a XMAS release.

**Comment by pepeEL** [comment]
but how we can use it ? please explaine me for gateway and node...

**Comment by tekka007** [comment]
@pepeEL since this PR is not merged yet, most probably the simplest solution for you is to download the repo here: https://github.com/tekka007/MySensors/tree/transportSMupdate

**Comment by pepeEL** [comment]
@tekka007 but how then use it in sketch gateway and node sketch ?

**Comment by tekka007** [comment]
@pepeEL #define MY_TRANSPORT_RELAXED

**Comment by pepeEL** [comment]
Only add to sketch gateway and node string:
#define MY_TRANSPORT_RELAXED

**Comment by pepeEL** [comment]
I test it and not working for me. I add Mysernsors Gateway Serial to my RPI2 with Domoticz. I write sketch from 2.0.1 beta and with add #define MY_TRANSPORT_RELAXED
I also upload sketch to my NODE also with new beta and added #define MY_TRANSPORT_RELAXED

Then i connect to my gateway. Sketch is for relay. I enable relay by domoticz then turn off my Gateway and manual change status relay to OFF by button on relay. Then i anable Gateway in Domoticz and status relay is still show ON, No update to OFF as i change manual.

Please help me....

**Comment by pepeEL** [comment]
Can anybody explaine me ?

**Comment by henrikekblad** [comment]
It's only in the node you need the define the MY_TRANSPORT_RELAXED. 

Blinking has not been removed from what I know. 

If something happens in the node while the gateway is turned off, the controller won't know about this. You could let your node periodically send a current status refresh to cover this problem. So when gateway comes up the controller will receive the updated status.

**Comment by pepeEL** [comment]
""If something happens in the node while the gateway is turned off, the controller won't know about this. You could let your node periodically send a current status refresh to cover this problem. So when gateway comes up the controller will receive the updated status.""



How i can set this ?

And what give me this function: MY_TRANSPORT_RELAXED  ?

**Comment by henrikekblad** [comment]
I'm sorry, but the MySensors library PR is not a good place to have a programming course. Please resort to forum.arduino.cc or forum.mysensors.org.

**Comment by tekka007** [comment]
Jenkins, build examples please

File: core/MyTransport.cpp [hunk 1]
```diff
@@ -89,22 +91,22 @@ void stInitUpdate(void) {
...
-			_nc.distance = 0u;
-			_nc.nodeId = GATEWAY_ADDRESS;
+			_transportConfig.parentNodeId = GATEWAY_ADDRESS;
+			_transportConfig.distanceGW = 0u;
+			_transportConfig.nodeId = GATEWAY_ADDRESS;
 			transportSetAddress(GATEWAY_ADDRESS);
 			// GW mode: skip FPAR,ID,UPL states
 			transportSwitchSM(stReady);
 		#else
 			if ((uint8_t)MY_NODE_ID != AUTO) {
```
Comment (henrikekblad) [inline core/MyTransport.cpp]: #if MY_NODE_ID != AUTO  perhaps?

**Pull request merged**